src/components/themed/CryptoExchangeFlipInput.tsx

```diff
@@ -0,0 +1,193 @@
+import { add } from 'biggystring'
+import { EdgeCurrencyWallet, EdgeDenomination } from 'edge-core-js'
+import * as React from 'react'
+import { useMemo, useState } from 'react'
+import { ActivityIndicator, View } from 'react-native'
+
+import { formatNumber } from '../../locales/intl'
+import { lstrings } from '../../locales/strings'
+import { useSelector } from '../../types/reactRedux'
+import { getTokenIdForced } from '../../util/CurrencyInfoHelpers'
+import { getWalletName } from '../../util/CurrencyWalletHelpers'
+import { convertNativeToDenomination } from '../../util/utils'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { CardUi4 } from '../ui4/CardUi4'
+import { CryptoIconUi4 } from '../ui4/CryptoIconUi4'
+import { RowUi4 } from '../ui4/RowUi4'
+import { EdgeText } from './EdgeText'
+import { ExchangedFlipInput2, ExchangedFlipInputAmounts } from './ExchangedFlipInput2'
+import { MainButton } from './MainButton'
+
+interface Props {
+  wallet?: EdgeCurrencyWallet
+  buttonText: string
+  headerText: string
+  currencyCode: string
+  displayDenomination: EdgeDenomination
+  overridePrimaryNativeAmount: string
+  isFocused: boolean
+  isThinking?: boolean
+  onFocuseWallet: () => void
+  onSelectWallet: () => void
+  onAmountChanged: (amounts: ExchangedFlipInputAmounts) => void
+  onNext: () => void
+  onFocus?: () => void
+  onBlur?: () => void
+  children?: React.ReactNode
+}
+
+export const CryptoExchangeFlipInput = (props: Props) => {
+  const { children, currencyCode, displayDenomination, onNext, overridePrimaryNativeAmount, wallet } = props
+
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  //
+  // State
+  //
+
+  const [errorMessage, setErrorMessage] = useState('')
+
+  //
+  // Derived State
+  //
+
+  const account = useSelector(state => state.core.account)
+
+  const tokenId = useMemo(() => {
+    if (wallet == null) return null
+    // This will error if wallet is undefined
+    return getTokenIdForced(account, wallet.currencyInfo.pluginId, currencyCode)
+  }, [account, currencyCode, wallet])
+
+  const cryptoAmount = useMemo(() => {
+    if (wallet == null || tokenId == null) return
+    const balance = wallet.balanceMap.get(tokenId) ?? '0'
+    const cryptoAmountRaw: string = convertNativeToDenomination(displayDenomination.multiplier)(balance)
+    return formatNumber(add(cryptoAmountRaw, '0'))
+  }, [displayDenomination.multiplier, tokenId, wallet])
+
+  const guiWalletName = useMemo(() => {
+    if (wallet == null) return ''
+    return getWalletName(wallet)
+  }, [wallet])
+
+  //
+  // Handlers
+  //
+
+  const handleAmountsChanged = (amounts: ExchangedFlipInputAmounts) => {
+    props.onAmountChanged(amounts)
+  }
```

Comment:
You can pass `props.onAmountChanged` straight down as a prop, without wrapping it in a handler. The ESLint rule has an exception for this case.